lib/MusicBrainz/Server/Data/Release.pm

```diff
@@ -302,17 +302,24 @@ sub find_by_release_group
       SELECT *
       FROM (
         SELECT DISTINCT ON (release.id) " . $self->_columns . ",
-          date_year, date_month, date_day, area.name AS country_name
+          date_year, date_month, date_day, area.name AS country_name,
+          rl.catalog_numbers AS catalog_numbers
         FROM " . $self->_table . "
         " . join(' ', @$extra_joins) . "
         LEFT JOIN release_event ON release_event.release = release.id
         LEFT JOIN area ON area.id = release_event.country
+        LEFT JOIN (
+          SELECT
+            array_agg(catalog_number ORDER BY catalog_number) AS catalog_numbers, release
+            FROM release_label
+            GROUP BY release
+        ) rl ON release.id = rl.release
         WHERE " . join(" AND ", @$conditions) . "
         ORDER BY release.id, date_year, date_month, date_day,
-          country_name, barcode
+          rl.catalog_numbers, country_name, barcode
```

Comment:
I'm not entirely sure what these are for if it sorts by `release.id` first, but

Reply:
That is fair, wonder whether there is any point at all to this order by here.

Reply:
Never mind, I overlooked the `SELECT DISTINCT ON` at first. The `ORDER BY` would seem relevant in that case: https://www.postgresql.org/docs/9.5/sql-select.html#SQL-DISTINCT